$MirOS: ports/print/cups/patches/patch-berkeley_lpq_c,v 1.3 2006/11/08 18:25:44 bsiegert Exp $
--- berkeley/lpq.c.orig	Sat Jul 12 00:48:26 2008
+++ berkeley/lpq.c	Tue Oct 21 22:56:22 2008
@@ -375,9 +375,9 @@ show_jobs(const char *command,		/* I - C
   if (dest == NULL)
   {
     if (id)
-      sprintf(resource, "ipp://localhost/jobs/%d", id);
+      snprintf(resource, sizeof(resource), "ipp://localhost/jobs/%d", id);
     else
-      strcpy(resource, "ipp://localhost/jobs");
+      strlcpy(resource, "ipp://localhost/jobs", sizeof(resource));
 
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI, "job-uri",
                  NULL, resource);
